kv/txn_coord_sender.go

```diff
@@ -309,6 +311,10 @@ func (tc *TxnCoordSender) Send(_ context.Context, call proto.Call) {
 	header := call.Args.Header()
 	tc.maybeBeginTxn(header)
 
+	header.CmdID = header.GetOrCreateCmdID(tc.clock.PhysicalNow())
+	tc.tracer.Begin(header, "coordsender_begin")
```

Comment:
I'm going to do some more work on the interface. Begin and Finalize are called when a request enters/leaves the system, respectively. They should attach the origin/location of where the request may go next, and I'll have to figure out whether we should continue allowing multiple calls to `Begin()` (they are counted against calls to `Finalize()`; this is necessary/useful since `DistSender` and `Store` share a tracer to combine as much of the trace as possible). Likely Finalize will lose its message. BTW, there's no special relationship between the Begin and Finalize calls in the final trace (i.e. no diff between them is stored directly, diffs are always between consecutive traced events), just in case that impression came up.